minigui/minigui.ts

```diff
@@ -0,0 +1,436 @@
+import * as board from './board'
+import * as util from './util'
+import {Log} from './log'
+import * as gtpsock from './gtp_socket'
+import {Graph} from './graph'
+
+const N = board.BoardSize.Nine;
+
+namespace BoardState {
+  export type Move = board.Point | 'pass' | 'resign';
+}
+
+class BoardState {
+  stones: Array<board.Color>;
+  principleVariation = new Array<board.Move>();
```

Comment:
I admire your commitment to princples ;)
https://en.wikipedia.org/wiki/Principal_variation_search
:rofl: (nit obvi)

Reply:
done